### PR TITLE
feat: enable trustyai to use different manifests path

### PR DESCRIPTION
**Pull request by zdtsw**
## Description

- trustyai has introduced overlays for odh and rhoai
- we can switch to use these than base

https://issues.redhat.com/browse/RHOAIENG-12277

## Merge criteria

- [ ] You have read the [contributors guide](https://github.com/opendatahub-io/opendatahub-operator/blob/incubation/CONTRIBUTING.md).
- [ ] Commit messages are meaningful - have a clear and concise summary and detailed explanation of what was changed and why.
- [ ] Pull Request contains a description of the solution, a link to the JIRA issue, and to any dependent or related Pull Request.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Review by ruivieira — APPROVED** [review]
LGTM, thanks @zdtsw !

File: components/trustyai/trustyai.go [hunk 1]
```diff
@@ -21,7 +21,9 @@ import (
 var (
 	ComponentName     = "trustyai"
 	ComponentPathName = "trustyai-service-operator"
-	Path              = deploy.DefaultManifestPath + "/" + ComponentPathName + "/base"
+	PathUpstream      = deploy.DefaultManifestPath + "/" + ComponentPathName + "/overlays/odh"
+	PathDownstream    = deploy.DefaultManifestPath + "/" + ComponentPathName + "/overlays/rhoai"
```
Comment (ykaliuta) [inline components/trustyai/trustyai.go:25]: Would be great if the paths were unified across the components that this definition could be kept in one place.

File: components/trustyai/trustyai.go [hunk 2]
```diff
@@ -45,7 +47,7 @@ func (t *TrustyAI) OverrideManifests(ctx context.Context, _ cluster.Platform) er
 		if manifestConfig.SourcePath != "" {
 			defaultKustomizePath = manifestConfig.SourcePath
 		}
-		Path = filepath.Join(deploy.DefaultManifestPath, ComponentPathName, defaultKustomizePath)
+		OverridePath = filepath.Join(deploy.DefaultManifestPath, ComponentPathName, defaultKustomizePath)
```
Comment (ykaliuta) [inline components/trustyai/trustyai.go:50]: Having such a global state is not the best pattern. It even can bring problems with parallel execution.
Better approach would be if the function returns the path (well, as a shortcut "" if it was not overridden, but some extra boolean looks nicer). Then the logic below could be the same, "if overridden, replace the default".

Comment (zdtsw) [reply components/trustyai/trustyai.go:50]: i am curious about this " _It even can bring problems with parallel execution_."
if you mean parallel as different components (trusty and kueue and so on) but the path variables are only defined within the package. or i missed something?

Comment (ykaliuta) [reply components/trustyai/trustyai.go:50]: > i am curious about this " _It even can bring problems with parallel execution_." if you mean parallel as different components (trusty and kueue and so on) but the path variables are only defined within the package. or i missed something?

Problems with params.env updating brings idea that we can encounter races when the same component is reconciled in parallel.

Comment (zdtsw) [reply components/trustyai/trustyai.go:50]: back to the reason i kept using this way, not change it to 
`func (d *type) OverrideManifests(ctx context.Context, _ cluster.Platform) (string, error) {`
is to keep the scope of this change only for Trusty.

we can do the change in another PR, to get OverrideManifests re-write for other components as well.
It was the same reason, when i did the change for dashboard months ago, did not want to mix refactor code.
then, as usual, never occured to me afterwards :laughing:

File: components/trustyai/trustyai.go (hunk 1 above)
Comment (zdtsw) [reply components/trustyai/trustyai.go:25]: it is very much up to each component how they would like to have their manifests.
e.g some from DW they only use "rhoai" for different "platform"
in trustyai's case, the componentPathName is not the same as ComponentName.
and some even have a dependentcomponent, which adds more complacity

**Review by ykaliuta — APPROVED** [review]

**Pull request merged**